packages/@webex/plugin-cc/src/AgentConfigService/AgentConfigService.ts

```diff
@@ -0,0 +1,126 @@
+import {WebexSDK, HTTP_METHODS} from '../types';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from './types';
+import Request from '../request';
+
+export default class AgentConfigService {
+  ciUserId: string;
+  orgId: string;
+  webex: WebexSDK;
+  wccAPIURL: string;
+  requestInstance: Request;
+
+  constructor(ciUserId: string, orgId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.ciUserId = ciUserId;
+    this.orgId = orgId;
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+    this.requestInstance = new Request(this.webex);
+  }
+
+  /**
+   * Method to get User using CI.
+   * @returns {Promise<UserResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getUserUsingCI(): Promise<UserResponse> {
+    try {
+      const URL = `${this.wccAPIURL}organization/${this.orgId}/user/by-ci-user-id/${this.ciUserId}`;
+      const response = await this.requestInstance.request(URL, HTTP_METHODS.GET);
+
+      this.webex.logger.log('getUserUsingCI api called successfully.');
+
+      return Promise.resolve(response?.body);
+    } catch (error) {
+      return Promise.reject(new Error(`Error while calling getUserUsingCI API, ${error}`));
+    }
+  }
+
+  /**
+   * Method to get Desktop Profile by passing Org Id.
+   * @param {string} desktopProfileId ID of the Desktop Profile to be retrieved.
+   * @returns {Promise<DesktopProfileResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getDesktopProfileById(desktopProfileId: string): Promise<DesktopProfileResponse> {
```

Comment:
This parameter is optional from where it is being invoked so shouldn't we avoid triggering unnecessary API requests if profileID was undefined?

Reply:
I updated the code and today in the scrum we had discussion around this, So the developer portal API does not mentioned anything around getting undefined value. So, I am calling getDesktopProfileById method by not passing optional parameter. The parameter (desktopProfileId) is mandatory.